app/models/youth_risk_assessment.rb

```diff
@@ -13,4 +15,8 @@ def self.framework_name
   def previous_assessment
     @previous_assessment ||= profile&.person&.latest_youth_risk_assessment
   end
+
+  def move_from_location
+    errors.add(:move, "'from_location' must be from either a secure training centre or a secure children's home") unless move&.from_location&.secure_training_centre? || move&.from_location&.secure_childrens_home?
```

Comment:
It's not possible to create a move without a from_location (it's a mandatory field) so this could possibly be simplified with fewer `&` 🤔

Reply:
If a move is indeed empty, all validations will run irrespective on the youth assessment. If we remove those checks we will get an error of `from_location` being called on `nil`. I've added a test as suggested below to confirm the behaviour and to test against it, making sure we get the correct validation error messages